src/main/java/de/embl/cba/mobie/ui/viewer/SourcesPanel.java

```diff
@@ -146,7 +145,7 @@ public void setSourceColor( String sourceName, Color color )
             return;
         }
 
-        final SourceAndMetadata< ? > sam = getSourceAndMetadata( sourceName );
+        final SourceAndMetadata< ? > sam = sourceNameToSourceAndMetadata.get(sourceName);
```

Comment:
Here you directly access:

`sourceNameToSourceAndMetadata.get(...)`

and below...